homeassistant/components/manual_mqtt/alarm_control_panel.py

```diff
@@ -446,11 +497,46 @@ async def message_received(msg):
             self.hass, self._command_topic, message_received, self._qos
         )
 
+        config = {
+            "version": 1,
+            "code_arm_required": self._code_arm_required,
+            "code_disarm_required": (self._code is not None),
+            "state_topic": self._state_topic,
+            "status_topic": self._status_topic,
+            "command_topic": self._command_topic,
+            "delay_times": self._delay_time_by_state,
+            "arming_times": self._arming_time_by_state,
+            "trigger_times": self._trigger_time_by_state,
+        }
+
+        def default(obj):
+            if isinstance(obj, (datetime.timedelta)):
+                return int(obj.total_seconds())
+            return str(obj)
+
+        mqtt.async_publish(
+            self.hass,
+            self._config_topic,
+            json.dumps(config, default=default),
+            self._qos,
+            True,
+        )
+
     async def _async_state_changed_listener(self, event):
         """Publish state change to MQTT."""
         new_state = event.data.get("new_state")
         if new_state is None:
             return
-        mqtt.async_publish(
-            self.hass, self._state_topic, new_state.state, self._qos, True
-        )
+        new_state_str = new_state.state
+        if new_state_str == "arming":
+            if new_state.attributes["next_state"] == "armed_away":
+                new_state_str = "arming_away"
+            elif new_state.attributes["next_state"] == "armed_home":
+                new_state_str = "arming_home"
+            elif new_state.attributes["next_state"] == "armed_night":
+                new_state_str = "arming_night"
+            else:
+                _LOGGER.warning(
+                    "Unknown arming next state: %s", new_state.attributes["next_state"]
+                )
+        mqtt.async_publish(self.hass, self._state_topic, new_state_str, self._qos, True)
```

Comment:
Please remove this change. I don't see why we should change this.

Reply:
> Please remove this change. I don't see why we should change this.

This is giving more information to the panel, so that the panel knows the target (planned) state.  This allows the panel to do something different based on whether you are arming into home or away (or night) states.

I can certainly remove the log warning, if that's your concern?

Reply:
We shouldn't change how we publish state.

Reply:
Okay, then how is a panel supposed to know, if it gets "arming", whether it is going into armed_home, armed_away, or armed_night mode?  This distinction is important so the panel can properly display the correct delay time with a countdown timer -- which it can do because it knows the delay times.  But this only works if it knows the arming target.

There is not enough additional information in the existing protocol.  For example, with this change I can arm it via the lovelace interface and my panels will react correctly, without it the panel doesn't know what to do -- so I proposed to add it here by differentiating `arming` into three "sub classes" to pass that information.

Do you have a suggestion for a better way to tell the alarm panels, via MQTT, the arming target?  I'm sure @thanksmister will be happy to change it too if we find a better way.

Reply:
@derekatkins As mentioned before, this change results into _wrong_ payloads being published over the `STATE_TOPIC`.
I refer to them as _wrong_, because the payloads don't match with the actual state of the `alarm_control_panel` entity and they are hard-coded in your PR.

I suggested 4 alternatives in my [earlier comment](https://github.com/home-assistant/core/pull/40973#issuecomment-735261604) which are more graceful, so its not like there is no choice here.
This PR is dangerous since it will (irreversibly) break stuff for everyone else and this change doesn't seem necessary nor an improvement to me.
I suggest to stick to the [state definition](https://developers.home-assistant.io/docs/core/entity/alarm-control-panel/#states) as used in all `alarm_control_panel` integrations in HA.

Reply:
@nielsfaber Please keep in mind that state changes can occur in two ways in this integration.  One is via the MQTT command topic, however it can also occur via the `alarm_control_panel` lovelace interface.  When the latter occurs, there is no MQTT reflection of the state change (i.e., the lovelace doesn't publish to the command topic).  So you're #1 suggestion, logic, can't work.
Your #2 option could work, but it seems overkill to send data twice, don't you think?  We're already sending the state, it's just we want to disambiguate.
Making the payloads configurable is certainly doable, and was on my (personal) roadmap, but I was trying to keep this change smaller and simpler.
Finally, I was trying to keep the MQTT protocol as close as possible to the existing.  I even made the code / json payload backwards compatible in order to try to not break current panels.  I understand that changes will require the users to rework their panel implementations, which is why I wanted to make breaking changes ONCE.  All the other work I have planned works within the protocol already defined and would be 100% compatible with this proposed protocol.